src/Features/Core/Portable/ConflictMarkerResolution/AbstractConflictMarkerCodeFixProvider.cs

```diff
@@ -45,91 +45,171 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             var cancellationToken = context.CancellationToken;
             var document = context.Document;
+
             var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
             var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var lines = text.Lines;
 
-            var startTrivia = root.FindTrivia(context.Span.Start);
-            if (!IsConflictMarker(text, startTrivia, '<'))
+            var position = context.Span.Start;
+            var conflictLine = lines.GetLineFromPosition(position);
+            if (position != conflictLine.Start)
+            {
+                Debug.Assert(false, "All conflict markers should start at the beginning of a line.");
                 return;
+            }
```

Comment:
This check is redundant, is part of `ShouldFix` too.

Reply:
true!